zerver/lib/streams.py

```diff
@@ -897,10 +899,11 @@ def get_streams_for_user(
 
     include_public = include_public and user_profile.can_access_public_streams()
 
-    # Start out with all active streams in the realm.
-    query = Stream.objects.filter(realm=user_profile.realm, deactivated=False)
+    # Start out with all streams in the realm.
+    query = Stream.objects.filter(realm=user_profile.realm)
 
     if include_all_active:
+        query = query.filter(deactivated=False)
```

Comment:
Hmm, this logic seems off for the case that one passed `include_all_active` and `include_archived`; I feel like that combination has to indicate they want all of them.

I think the logic we want is instead just for deactivated streams to be filtered, unless `include_archived` is passed, in which case they aren't -- i.e. a meaning of "don't exclude archived" rather than working like the other options. 

It's worth flagging this detail in the API design discussion though.

Reply:
Yeah I improved the logic.